json/src/main/java/com/basistech/rosette/apimodel/jackson/recordsimilaritydeserializers/RecordSimilarityDeserializerUtilities.java

```diff
@@ -33,9 +35,23 @@
 
 final class RecordSimilarityDeserializerUtilities {
 
-    private RecordSimilarityDeserializerUtilities() { }
+    private RecordSimilarityDeserializerUtilities() {
+    }
+
+    public static RecordSimilarityResult parseResult(JsonNode node, JsonParser jsonParser, Map<String, RecordSimilarityFieldInfo> fields) throws IOException {
+        final Double score = node.get("score") != null ? node.get("score").traverse(jsonParser.getCodec()).readValueAs(Double.class) : null;
+        final RecordSimilarityExplainInfo explainInfo = node.get("explainInfo") != null ? node.get("explainInfo").traverse(jsonParser.getCodec()).readValueAs(RecordSimilarityExplainInfo.class) : null;
+        Map<String, RecordSimilarityField> left = node.get("left") != null && fields != null ? parseRecord(node.get("left"), jsonParser, fields) : null;
+        Map<String, RecordSimilarityField> right = node.get("right") != null && fields != null ? parseRecord(node.get("right"), jsonParser, fields) : null;
+        return RecordSimilarityResult.builder()
```

Comment:
while we're in here already, please fix the related bug where `error` is not being parsed, i.e.
`final String error = node.get("error") != null ? node.get("error").traverse(jsonParser.getCodec()).readValueAs(String.class) : null;` or `String error = Optional.ofNullable(node.get("error")).map(JsonNode::asText).orElse(null);`